File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
def dfs(level,wo,word):
    global answer, arr,cnt
    if wo == word:
        cnt=answer
        return answer
    if level == 5:
        return
    
    for i in range(5):
        answer+=1
        new_wo = wo+arr[i]
        dfs(level+1, new_wo, word)

def solution(word):
    global answer,arr,cnt
    answer = 0
    cnt=0
    arr=['A','E','I','O','U']
    dfs(0,'',word)
    return cnt

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
   
function solution(word) {
    const word_alpha = ['A','E','I','O','U']

    let answer = 0;
    let checked = false;
    
    function dfs(currentWord) {
        if (currentWord === word) {
            checked = true;
            return;
        }
        if (currentWord.length >= 5 || checked) return;
        for (let alpha of word_alpha){
            if (checked) return;
            answer ++
            dfs(currentWord+alpha)
        }
    }
    dfs('')
    return answer;
}